### ISSUE TITLE
Way to compare image hashes

### ISSUE THREAD

**Issue by benkasminbullock** [runarbu/PerlImageHash#5]
At the moment the module returns image hashes but it doesn't seem to offer a way to compare the hashes of two different images. I request adding the feature of a simple way of comparing images.

**Comment by runarbu** [comment]
There are currently two ways to compare the hashes. If they are totally equal you can just do a 'eq', but the hashes rarely are totally equal. 

If the hash is slightly different you need to compare the Hamming distance like this:

        # Hamming distance. Take two hashes in. 
        # Returns the distance. If distance is below 5 the image is normally equal
        sub hammingdistance {
                my ($a, $b) = @_;

                my $distance = 0;
                for (my $i = 0; $i < 64; $i++) {
                        if ($a->{'hash'}->[$i] != $b->{'hash'}->[$i]) {
                                $distance++;
                        }
        
                }

                return $distance;
        }

The hammingdistance function sold definitely be added and this better documented.

Does this clarifies it, or did you mean on some other way of comparing images?

**Comment by benkasminbullock** [comment]
I think casual users of the module would find the module much more useful if there was a simple way to compare the hashes, or even an example in the documentation of how to compare the hashes.

**Comment by runarbu** [comment]
I agree, you are right. The hammingdistance function sold definitely be added and better documented.

Casual users may also be interested in using the hammingdistance function together with Tree::BK to easily do fuzzy matching on a large set of images, so that should be documented too.

**Comment by benkasminbullock** [comment]
If there is anything I can do to assist in getting this done, let me know.
